lib/ansible/modules/cloud/vmware/vmware_vm_facts.py

```diff
@@ -150,7 +150,9 @@
 class VmwareVmFacts(PyVmomi):
     def __init__(self, module):
         super(VmwareVmFacts, self).__init__(module)
-        self.custom_field_mgr = self.content.customFieldsManager.field
+        self.custom_field_mgr = []
+        if self.content.customFieldsManager:  # not an ESXi
```

Comment:
My fault with that initial code,
maybe it's better to try & except for AttributeError here?
```python
try:
    self.custom_field_mgr = self.content.customFieldsManager.field
except AttributeError:
    pass
```

I can't check it right now against ESXi, but this definitely would be nice to have.

Reply:
@goneri [`vmware_guest_custom_attributes`](https://github.com/ansible/ansible/blob/14ab205ea26b99e4a33a092dedd2f5d6ad1ae7c9/lib/ansible/modules/cloud/vmware/vmware_guest_custom_attributes.py#L142) and [`vmware_guest_custom_attribute_defs`](https://github.com/ansible/ansible/blob/14ab205ea26b99e4a33a092dedd2f5d6ad1ae7c9/lib/ansible/modules/cloud/vmware/vmware_guest_custom_attribute_defs.py#L94) has the same code, so it would be great to fix those as well.

I am OK with any implementation.

Reply:
Thanks @Akasurde, I moved the `cusom_field_mgr` field initialization in the constructor to avoid to much dup code.